civicaseextras.php

```diff
@@ -134,34 +136,81 @@ function civicaseextras_civicrm_entityTypes(&$entityTypes) {
   _civicaseextras_civix_civicrm_entityTypes($entityTypes);
 }
 
-function civicaseextras_civicrm_alterAngular(\Civi\Angular\Manager $angular) {
-  _civicaseextras_civicrm_alterAngular($angular);
+function civicaseextras_civicrm_alterAngular(AngularManager $angular) {
+  _civicaseextras_alterAngular_addVisualAlert($angular);
+  _civicaseextras_alterAngular_appendOutcomePanel($angular);
 }
 
-// --- Functions below this ship commented out. Uncomment as required. ---
+/**
+ * Replaces the date column type on case lists so it displays an alert when the
+ * given date is overdue.
+ *
+ * @param $angular AngularManager
+ */
+function _civicaseextras_alterAngular_addVisualAlert (AngularManager &$angular) {
+  $changeSet = AngularChangeSet::create('display_warning_for_overdue_cases')
+    ->alterHtml('~/civicase/case/list/directives/case-list-table.directive.html',
+      function (phpQueryObject $doc) {
+        $doc->find('[ng-switch-when="date"]')
+          ->html('
+            <span ng-if="!item.overdueDates[header.name]">
+              {{ CRM.utils.formatDate(item[header.name]) }}
+            </span>
+            <strong ng-if="item.overdueDates[header.name]"
+              class="text-danger">
+              {{ CRM.utils.formatDate(item[header.name]) }}
+              <i class="material-icons civicase__icon">error</i>
+            </strong>
+          ');
+      });
+  $angular->add($changeSet);
+}
 
 /**
- * Implements hook_civicrm_preProcess().
+ * Appends the case outcomes to the case details summary.
  *
- * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_preProcess
+ * @param $angular AngularManager
+ */
+function _civicaseextras_alterAngular_appendOutcomePanel (AngularManager &$angular) {
+  $changeSet = AngularChangeSet::create('inject_case_outcomes')
+    ->alterHtml('~/civicase/case/details/summary-tab/case-summary-custom-data.html',
```

Comment:
I dont see this mentioned as a requirement in the ticket, are you sure this is required?

Reply:
No, I moved the function from `civicaseextras.civix.php` to `civicaseextras.php` because I mistakenly added that function to the `civix` file (which is autogenerated).

Reply:
Also, the template path was outdated.